src/uct/cuda/cuda_ipc/cuda_ipc.cuh

```diff
@@ -178,34 +230,32 @@ void uct_cuda_ipc_copy_level<UCS_DEVICE_LEVEL_BLOCK>(void *dst, const void *src,
 {
     using vec4 = int4;
     using vec2 = int2;
-    auto s1  = reinterpret_cast<const char*>(src);
-    auto d1  = reinterpret_cast<char *>(dst);
-    const vec4 *s4;
-    vec4 *d4;
-    int warp, num_warps, idx;
+    auto      s1        = reinterpret_cast<const char*>(src);
+    auto      d1        = reinterpret_cast<char *>(dst);
+    const int warp      = threadIdx.x / UCS_DEVICE_NUM_THREADS_IN_WARP;
+    const int num_warps = blockDim.x / UCS_DEVICE_NUM_THREADS_IN_WARP;
+    const int idx       = threadIdx.x % UCS_DEVICE_NUM_THREADS_IN_WARP;
     size_t num_lines;
 
     if (UCT_CUDA_IPC_IS_ALIGNED_POW2((intptr_t)s1, sizeof(vec4)) &&
         UCT_CUDA_IPC_IS_ALIGNED_POW2((intptr_t)d1, sizeof(vec4))) {
+        const vec4 *s4 = reinterpret_cast<const vec4*>(s1);
+        vec4       *d4 = reinterpret_cast<vec4*>(d1);
         vec4 tmp[UCT_CUDA_IPC_COPY_LOOP_UNROLL];
-        warp      = threadIdx.x / UCT_CUDA_IPC_WARP_SIZE;
-        num_warps = blockDim.x / UCT_CUDA_IPC_WARP_SIZE;
-        idx       = threadIdx.x % UCT_CUDA_IPC_WARP_SIZE;
-        s4        = reinterpret_cast<const vec4*>(s1);
-        d4        = reinterpret_cast<vec4*>(d1);
-        num_lines = (len / (UCT_CUDA_IPC_WARP_SIZE * UCT_CUDA_IPC_COPY_LOOP_UNROLL * sizeof(vec4))) *
-                    (UCT_CUDA_IPC_WARP_SIZE * UCT_CUDA_IPC_COPY_LOOP_UNROLL);
-
-        for (size_t line = warp * UCT_CUDA_IPC_WARP_SIZE * UCT_CUDA_IPC_COPY_LOOP_UNROLL + idx; line < num_lines;
-             line += num_warps * UCT_CUDA_IPC_WARP_SIZE * UCT_CUDA_IPC_COPY_LOOP_UNROLL) {
+
+        num_lines = (len / (UCS_DEVICE_NUM_THREADS_IN_WARP * UCT_CUDA_IPC_COPY_LOOP_UNROLL * sizeof(vec4))) *
+                    (UCS_DEVICE_NUM_THREADS_IN_WARP * UCT_CUDA_IPC_COPY_LOOP_UNROLL);
```

Comment:
maybe separate var for `UCS_DEVICE_NUM_THREADS_IN_WARP * UCT_CUDA_IPC_COPY_LOOP_UNROLL`?